tests/test_validation.py

```diff
@@ -0,0 +1,59 @@
+import unittest
+import pathlib
+
+import jsonschema
+import yaml
+
+from lsst.ts import salobj
+
+
+class ValidationTestCase(unittest.TestCase):
+    def setUp(self):
+        schemaname = "PMD.yaml"
+        pkg_dir = pathlib.Path(__file__).parents[1]
+        schemapath = pkg_dir / "schema" / schemaname
+        with open(schemapath, "r") as f:
+            rawschema = f.read()
+        self.schema = yaml.safe_load(rawschema)
+        self.validator = salobj.DefaultingValidator(schema=self.schema)
+
+    def test_all_specified(self):
+        data = {
+            "hub_config": [
+                {
+                    "sal_index": 1,
+                    "telemetry_interval": 1,
+                    "devices": ["Dial Gage"],
+                    "units": "um",
+                    "location": "Office",
+                    "serial_port": "/dev/ttyUSB0",
+                    "hub_type": "Mitutoyo",
+                }
+            ]
+        }
+        data_copy = data.copy()
+        result = self.validator.validate(data)
+        self.assertEqual(data, data_copy)
+        for field, value in data.items():
+            self.assertEqual(result[field], value)
+
+    def test_invalid_configs(self):
+        data = {
+            "hub_config": [
+                {
+                    "sal_index": 0,
+                    "telemetry_interval": "Sandwich",
+                    "devices": 1,
+                    "units": "fff",
+                    "location": 1,
+                    "serial_port": 12,
+                    "hub_type": "Sandwich",
+                }
+            ]
+        }
```

Comment:
I strongly suggest making this test use each invalid entry separately. That makes this a much more powerful test. As it stands you are only testing that one of these many incorrect items is causing a failure to validate, and it could be any of them.

For example:
```
good_data = {...same data you used in test_all_specified...}
bad_hub_items = dict(
    sal_index=0,
    telemetry_interval="not a number",
    ....
}
for name, bad_value in bad_hub_items.items():
    bad_data = good_data.copy()
    bad_data["num_config"][name] = bad_value
    with self.assertRaises(jsonschema.exceptions.ValidationError):
        self.validator.validate(bad_data)
```
If you ever add non-hub_config configuration items then you can add a new section to this test that specifies invalid values for those items.